docs/guides/advanced-use-case/cross-namespace-repository.md

```diff
@@ -0,0 +1,689 @@
+
+> For GCS backend, if the bucket does not exist, Stash needs `Storage Object Admin` role permissions to create the bucket. For more details, please check the following [guide](/docs/guides/backends/gcs.md).
+
+**Create Secret:**
+
+Let's create a secret called `gcs-secret` with access credentials to our desired GCS bucket,
+
+```bash
+$ echo -n 'changeit' > RESTIC_PASSWORD
+$ echo -n '<your-project-id>' > GOOGLE_PROJECT_ID
+$ cat /path/to/downloaded-sa-key.json > GOOGLE_SERVICE_ACCOUNT_JSON_KEY
+$ kubectl create secret generic -n stash gcs-secret \
+    --from-file=./RESTIC_PASSWORD \
+    --from-file=./GOOGLE_PROJECT_ID \
+    --from-file=./GOOGLE_SERVICE_ACCOUNT_JSON_KEY
+secret/gcs-secret created
+```
+
+Now, we are ready to backup our workload's data to our desired backend.
+
+**Create Repository:**
+
+Now, create a `Repository` using this secret. Below is the YAML of `Repository` crd we are going to create. Notice the `spec.usagePolicy` section. Here, we are allowing all namespaces to refer to this repository. You can restrict this capability to a particular namespace. Please, check the following guide from here for more details.
+
+```yaml
+apiVersion: stash.appscode.com/v1alpha1
+kind: Repository
+metadata:
+  name: gcs-repo
+  namespace: stash
+spec:
+  backend:
+    gcs:
+      bucket: stash-testing
+      prefix: /cross-namespace/data/sample-statefulset
+    storageSecretName: gcs-secret
+  usagePolicy:
+    allowedNamespaces:
+      from: All
+```
+
+Let's create the Repository we have shown above,
+```bash
+$ kubectl apply -f https://github.com/stashed/docs/raw/{{< param "info.version" >}}/docs/examples/guides/advanced-use-case/cross-namespace-backup/repository.yaml
+repository.stash.appscode.com/gcs-repo created
+```
+Now, we are ready to backup our sample data into this backend.
+
+### Backup
+
+We are going to create a `BackupConfiguration` crd in `prod` namespace targeting the `stash-demo` StatefulSet that we have deployed earlier. This `BackupConfiguration` will refer to the `gcs-repo` repository of `stash` namespace. Stash will inject a sidecar container into the target. It will also create a `CronJob` to take periodic backup of the `/source/data` directory of the target.
+
+**Create BackupConfiguration:**
+
+Below is the YAML of the `BackupConfiguration` crd that we are going to create,
+
+```yaml
+apiVersion: stash.appscode.com/v1beta1
+kind: BackupConfiguration
+metadata:
+  name: ss-backup
+  namespace: prod
+spec:
+  repository:
+    name: gcs-repo
+    namespace: stash
+  schedule: "*/5 * * * *"
+  target:
+    ref:
+      apiVersion: apps/v1
+      kind: StatefulSet
+      name: stash-demo
+    volumeMounts:
+    - name: source-data
+      mountPath: /source/data
+    paths:
+    - /source/data
+  retentionPolicy:
+    name: 'keep-last-5'
+    keepLast: 5
+    prune: true
+```
+
+Here,
+
+- `spec.repository.name` refers to the `Repository` object `gcs-repo` that holds backend information.
```

Comment:
refers to the name of the `Repository` object that holds the backend information